hs_core/authentication.py

```diff
@@ -37,10 +40,22 @@ def build_oidc_url(request):
     Returns:
         string: redirect URL for oidc service
     """
-    view, args, kwargs = resolve(reverse('oidc_authentication_init'))
+    oidc_init = reverse('oidc_authentication_init')
+    view, args, kwargs = resolve(oidc_init)
     kwargs["request"] = request
-    redirect = view(*args, **kwargs)
-    return redirect.url
+    error = None
+    try:
+        redirect = view(*args, **kwargs)
+        if isinstance(redirect, Exception):
+            error = redirect
+        else:
+            return redirect.url
+    except Exception as e:
+        error = e
+
+    # If the OIDC URL could not be built, log the error and return the bare oidc_authentication_init url
+    logger.error(f"Error building OIDC URL: {error}")
+    return oidc_init
```

Comment:
this will cause the user to be redirected to the **signin page** (instead of the signup page), in the case of error.